wrangler-transform/src/main/java/co/cask/wrangler/Wrangler.java

```diff
@@ -280,6 +286,69 @@ public void initialize(TransformContext context) throws Exception {
     errorCounter = 0;
   }
 
+  /**
+   * Prepares Wrangler before execution of transform. Mainly preparing and sending field-level metadata
+   *
+   * @param context framework for transform stages
+   */
+  @Override
+  public void prepareRun(StageSubmitter<TransformContext> context) throws Exception {
+    if (context.getContext().getArguments().has("enable.wrangler.field.lineage")) {
+      List<String> startColumns, endColumns;
+      Schema inputSchema, outputSchema;
+      inputSchema = context.getContext().getInputSchema();
+      outputSchema = context.getContext().getOutputSchema();
+      if (inputSchema == null || inputSchema.getFields() == null) {
+        startColumns = new ArrayList<>(0);
+      } else {
+        startColumns = new ArrayList<>(inputSchema.getFields().size());
+        for (Schema.Field field : inputSchema.getFields()) {
+          startColumns.add(field.getName());
+        }
+      }
+      if (outputSchema == null || outputSchema.getFields() == null) {
+        endColumns = new ArrayList<>(0);
+      } else {
+        endColumns = new ArrayList<>(outputSchema.getFields().size());
+        for (Schema.Field field : outputSchema.getFields()) {
+          endColumns.add(field.getName());
+        }
+      }
+      FieldLevelLineage f;
+      store = new DefaultTransientStore();
+      registry = new CompositeDirectiveRegistry(
+        new SystemDirectiveRegistry(),
+        new UserDirectiveRegistry(context.getContext())
+      );
+      RecipeParser directives = new GrammarBasedParser(new MigrateToV2(config.directives).migrate(), registry);
+      ExecutorContext ctx = new WranglerPipelineContext(ExecutorContext.Environment.TRANSFORM,
+                                                        context.getContext(), store);
+      try {
+        // Create the pipeline executor with context being set.
+        pipeline = new RecipePipelineExecutor();
+        pipeline.initialize(directives, ctx);
+      } catch (Exception e) {
+        throw new Exception(
+          String.format("Stage:%s - %s", context.getContext().getStageName(), e.getMessage())
```

Comment:
You're still passing `e` as the second param to String.format, and not as the second parameter of the Exception constructor.